graphql_compiler/compiler/emit_sql.py

```diff
@@ -851,7 +865,18 @@ def _join_to_parent_location(
                 f"during fold {self}."
             )
 
-        self._came_from[self._current_alias] = self._current_alias.c[to_column]
+        # construct on clause for join
+        if isinstance(join_descriptor, DirectJoinDescriptor):
+            matching_column_pairs = {
+                (join_descriptor.from_column, join_descriptor.to_column),
+            }
+        elif isinstance(join_descriptor, CompositeJoinDescriptor):
+            matching_column_pairs = join_descriptor.column_pairs
+        else:
+            raise AssertionError(u"TODO")
+
+        non_null_column = sorted(matching_column_pairs)[0][1]
```

Comment:
The `[0][1]` seems a bit less than ideal, care to add a comment to it?

Is it ever possible that `matching_column_pairs` is empty? I'm assuming its type is something like `AbstractSet[Tuple[str, str]]` so the `[1]` should always be safe.